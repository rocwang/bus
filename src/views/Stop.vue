<template>
  <PanelStop
    :stopCode="stopCode"
    :stopName="stopName"
    :routeShortNames="routeShortNames"
  />
</template>

<script>
import PanelStop from "../components/PanelStop";
import { stopCode$, stopName$ } from "../store/stop";
import { actionViewStop$ } from "../store/actions";
import { routeShortNamesByStop$ } from "../store/routes";

export default {
  name: "Stop",
  components: { PanelStop },
  subscriptions() {
    return {
      stopName: stopName$,
      routeShortNames: routeShortNamesByStop$,
      stopCode: stopCode$
    };
  },
  beforeRouteEnter: (to, from, next) => {
    actionViewStop$.next(to.params.stopCode);
    next();
  },
  beforeRouteUpdate(to, from, next) {
    actionViewStop$.next(to.params.stopCode);
    next();
  },
  beforeRouteLeave(to, from, next) {
    actionViewStop$.next("");
    next();
  }
};
</script>
