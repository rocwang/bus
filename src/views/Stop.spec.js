import { shallowMount } from "@vue/test-utils";
import Stop from "./Stop";
import "fake-indexeddb/auto";

describe("Stop.vue", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(Stop, {
      mocks: {
        $route: {
          params: {
            stopCode: "1234"
          }
        }
      }
    });
  });

  it("matches the snapshot", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it.skip("triggers the view stop action before route enter", () => {});
  it.skip("triggers the view stop action before route update", () => {});
  it.skip("triggers the view stop action before route leave", () => {});
});
