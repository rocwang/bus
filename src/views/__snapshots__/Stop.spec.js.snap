// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stop.vue matches the snapshot 1`] = `
<panelstop-stub
  routeshortnames="NX1"
  stopcode="1234"
  stopname="ABCD"
/>
`;
